### PR TITLE
Copy URL to clipboard when share button is clicked

### PR DESCRIPTION
**Pull request by goldenapples**
Provides a minimally styled share button which can be wired up by adding the class name "is-share-button" to a button.

If this class is set on the button, clicking it will copy the current URL to the user's clipboard and briefly update the button text to say "Copied!".

image

### CODE REVIEW

File: src/styles/core-buttons.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.wp-block-button {
+
+	&.is-share-button {
+		.wp-element-button {
+			cursor: pointer !important;
+			transition: background-color 200ms ease-in;
+			position: relative;
+			pinter-events: all;
```
Comment (kadamwhite) [inline src/styles/core-buttons.scss:8]: Typo
```suggestion
			pointer-events: all;
```

File: src/styles/core-buttons.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+			&:after {
+				position: absolute;
+				left: 0;
+				right: 0;
+				top: 0;
+				bottom: 0;
+				background-color: var( --wp--preset--color--wmf-report-green, #318557 ) !important;
+				color: #fff;
+				content: "Copied!";
```
Comment (kadamwhite) [inline src/styles/core-buttons.scss:21]: Not necessary for this project but a trick I like to use is to use a filter to write a data-attr onto the element with localized text, which we can then use in the `content:` rule with the `attr()` css function. Not needed this time though!

**Review by kadamwhite — APPROVED** [review]
Really appreciate you grabbing this, Than!

File: src/styles/core-buttons.scss (hunk 2 above)
Comment (goldenapples) [reply src/styles/core-buttons.scss:21]: oh, that's good. I'll use that in the future.

File: src/blocks/overlay/render.php [hunk 3]
```diff
@@ -7,4 +7,10 @@
 
 <template <?php echo get_block_wrapper_attributes( [ 'class' => 'alignwide' ] ); ?>>
 	<?php echo $content; ?>
+	<!-- wp:buttons -->
+	<div class="wp-block-buttons">
+	<!-- wp:button {"backgroundColor":"base100","className":"is-style-transparent is-share-button"} -->
+	<div class="wp-block-button is-style-transparent is-share-button"><a class="wp-block-button__link has-base-100-background-color has-background wp-element-button">Share This</a></div>
+	<!-- /wp:button --></div>
+	<!-- /wp:buttons -->
```
Comment (goldenapples) [inline src/blocks/overlay/render.php:15]: shame.html

**Review by kadamwhite — APPROVED** [review]

**Pull request merged**